src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -85,6 +96,34 @@ private String convertKey(Schema keySchema, Object key) {
   }
 
   public IndexableRecord convertRecord(SinkRecord record, String index, String type, boolean ignoreKey, boolean ignoreSchema) {
+    if (record.value() == null) {
+      switch (behaviorOnNullValues) {
+        case IGNORE:
+          return null;
+        case DELETE:
+          // Will proceed as normal, ultimately creating an IndexableRecord with a null payload
+          break;
```

Comment:
Note that the behavior is described by the config description as "How to handle records with a non-null key and a null value", but this logic is not really checking whether the key is not null. Do we need to do something like this:

```
    case DELETE:
      if (!ignoreKey && record.key() == null) {
        // We should use the key but the key is also null, so ignore
        return null;
      }
      // Will proceed as normal, ultimately creating an IndexableRecord with a null payload
      break;
```

Reply:
Should they be ignored even when `ignoreKey` is `true`, then?

Reply:
No, they should not be ignored when `ignoreKey == true` because then there is a key created from the topic partition and offset information.

Reply:
Aren't t/p/o guaranteed to be unique per message?

Reply:
Good point. If a record's key and value are both null, then DELETE behavior can't do anything with the null and DELETE means nothing for a generated key. Just add a comment to clarify this.